### PR TITLE
Add pytorch container

### PR DESCRIPTION
**Pull request by surajssd**

### CODE REVIEW

**Comment by surajssd** [comment]
I think the image can be optimised a bit to reduce its size. But we can work on it once we are sure that the current state of the image is workable for us.

**Comment by t-lo** [comment]
Ran into issues with the container build on x86; `libGL` is missing, and the from-sources numpy installation breaks the benchmark on x86. I think we need a branch in the Dockerfile to install libGL and not install numpy from scratch for x86.

Some minor ideas:
- can we add `echo -e '[pytest]\nconsole_output_style=progress\nlog_cli=true' > pytest.ini` at the end of the Dockerfile? It will make the benchmark emit runtime status information to stdout.
- can we add a convenience wrapper script which sources the venv and then execs `python test_bench.py $@`?

**Comment by surajssd** [comment]
@t-lo PTAL.

File: pytorch/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,73 @@
+FROM fedora
+
+RUN dnf install -y \
```
Comment (t-lo) [inline pytorch/Dockerfile:3]: Can we add `python` here so there is a plain `python` executable?

**Comment by t-lo** [comment]
Fails in various places because a plain `python` executable is missing. Fix proposed in code.

File: pytorch/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+RUN cd && \
+    virtualenv -p 3.7 venv && \
+    source venv/bin/activate && \
+    python -m pip install --upgrade pip && \
+    pip install \
+        torch \
+        torchtext \
+        torchvision
+
+RUN cd && \
```
Comment (t-lo) [inline pytorch/Dockerfile:26]:
```suggestion
RUN cd && \
    source venv/bin/activate && \
```

File: pytorch/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        git checkout v0.5.0 && \
+        git clean -fd && \
+        mkdir build && \
+        cd build && \
+        cmake .. && \
+        make -j4 && \
+        cd ../python && \
+        python setup.py install
+
+RUN cd && \
```
Comment (t-lo) [inline pytorch/Dockerfile:62]:
```suggestion
RUN cd && \
    source venv/bin/activate && \
```

File: pytorch/run.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -xeuo pipefail
+
+source /root/venv/bin/activate
+
+cd /root/benchmark
+python3 test_bench.py $@
```
Comment (t-lo) [inline pytorch/run.sh:8]:
```suggestion
exec python test_bench.py $@
```

File: pytorch/Dockerfile (hunk 2 above)
Comment (surajssd) [reply pytorch/Dockerfile]: Oops I saw these comments later, but pushed the changes already.

**Review by t-lo — APPROVED** [review]
LGTM, thank you!

**Pull request merged**